### ISSUE TITLE
Fetching PGP key and having no correct GnuPg setup

### ISSUE THREAD

**Issue by SteveClement** [MISP/MISP#2891]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | ubuntu
| MISP version / git hash   |v2.4.87 (72d0c61ceab2578bba673a2b1cd5ed92e193a09c)	

### Expected behaviour

When editing a user profile and fetching a GPG key without proper GPG setup, makes MISP behave unexpectedly.

1. Saving your changes in Request URL: /users/view/1 will work but you will  get an error, see Screenshot and debug message [0]
2. Clicking on username to get Request URL: /users/view/me results in an Internal Error due to the same GPG/PGP unconfiguration.
3. no more able to access the user view to change or remove the key or change email etc.

### Logs, screenshots, configuration dump, ...

Debug message [0]
```
MISPkappa/app/tmp/logs# cat error.log
2018-01-30 10:05:20 Error: The 'homedir' "" is not readable or does not exist and cannot be created. This can happen if 'homedir' is not specified in the Crypt_GPG options, Crypt_GPG is run as the web user, and the web user has no home directory.
2018-01-30 10:05:20 Error: [Crypt_GPG_FileException] The 'homedir' "" is not readable or does not exist and cannot be created. This can happen if 'homedir' is not specified in the Crypt_GPG options, Crypt_GPG is run as the web user, and the web user has no home directory.
Request URL: /users/view/1
Stack Trace:
#0 /var/www/MISPkappa/app/Vendor/pear/crypt_gpg/Crypt/GPGAbstract.php(288): Crypt_GPG_Engine->__construct(Array)
#1 /var/www/MISPkappa/app/Model/User.php(475): Crypt_GPGAbstract->__construct(Array)
#2 /var/www/MISPkappa/app/Controller/UsersController.php(54): User->verifySingleGPG(Array)
#3 [internal function]: UsersController->view('1')
#4 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(UsersController), Array)
#5 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#6 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(UsersController), Object(CakeRequest))
#7 /var/www/MISPkappa/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#8 {main}
```

Debug message [1]
```
2018-01-30 10:11:29 Error: [Crypt_GPG_FileException] The 'homedir' "" is not readable or does not exist and cannot be created. This can happen if 'homedir' is not specified in the Crypt_GPG options, Crypt_GPG is run as the web user, and the web user has no home directory.
Request URL: /users/view/me
Stack Trace:
#0 /var/www/MISPkappa/app/Vendor/pear/crypt_gpg/Crypt/GPGAbstract.php(288): Crypt_GPG_Engine->__construct(Array)
#1 /var/www/MISPkappa/app/Model/User.php(475): Crypt_GPGAbstract->__construct(Array)
#2 /var/www/MISPkappa/app/Controller/UsersController.php(54): User->verifySingleGPG(Array)
#3 [internal function]: UsersController->view('1')
#4 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(UsersController), Array)
#5 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#6 /var/www/MISPkappa/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(UsersController), Object(CakeRequest))
#7 /var/www/MISPkappa/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#8 {main}
```

![screen shot 2018-01-30 at 10 05 44](https://user-images.githubusercontent.com/362025/35557566-c0fd2f0a-05a5-11e8-89bb-00241698a808.png)

**Comment by iglocska** [comment]
Interesting, that shouldn't happen. Will have a look.

Also, MISPkappa?...

![image](https://user-images.githubusercontent.com/3668672/35557996-1f3f3e7c-05a7-11e8-9ceb-c4d28d3c5674.png)

**Comment by JakubOnderka** [comment]
Should be fixed after #5120 is merged.
